src/core/nem/external/nacl-fast.js

```diff
@@ -811,6 +827,55 @@
       return 0;
     }
 
+    function unpack(r, p) {
```

Comment:
perhaps export `unpackneg` from js script rather than duplicating it?

Reply:
thanks, make sense 👍🏼

Reply:
I did a double-check on the code, compare with `unpackneg` vs `unpack` most of the code is the same but here is the different part.

in `unpackneg`
` if (par25519(r[0]) === (p[31]>>7)) Z(r[0], gf0, r[0]);`

in `unpack`
` if (par25519(r[0]) !== (p[31]>>7)) Z(r[0], gf0, r[0]);`

Reply:
ah dang, right unpackneg() = unpack AND negate, technically we could extract and/or parametrize this, but not sure if's good idea. cc @Jaguar0625

Reply:
yea, i'd recommend refactoring unpack and unpackneg

Reply:
@gimre-xymcity or could we negate after calling unpackneg?